torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -751,6 +773,24 @@ def _run_param_init_fn():
             if p not in ignored_params and not isinstance(p, FlatParameter)
         ]
 
+        if sync_module_states:
+            # Collect buffers we have to synchronize, avoiding buffers that have already
+            # been synchronized to avoid redundant synchronization.
+            bufs_to_sync = []
+            for buf in module.buffers():
+                if not getattr(buf, '_fsdp_has_been_sync', False):
+                    buf._fsdp_has_been_sync = True
+                    bufs_to_sync.append(buf.detach())
+
+            _sync_params_and_buffers(
+                process_group=self.process_group,
+                # TODO: FSDP should sync buffers as well
```

Comment:
Is this TODO fixed now since it seems that buffers are being synced?

Reply:
Yep, should be fixed